thingif/src/main/java/com/kii/thingif/command/AliasAction.java

```diff
@@ -1,19 +1,101 @@
 package com.kii.thingif.command;
 
-public class AliasAction<T extends Action> {
-    private String alias;
-    private T action;
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
 
-    public AliasAction(String alias, T action) {
+import com.google.gson.Gson;
+
+/**
+ * Represent Action of alias
+ * @param <T> Action class
+ */
+public class AliasAction<T extends Action> implements Parcelable{
+    @NonNull private String alias;
+    @NonNull private T action;
+
+    private volatile int hashCode; // cached hashcode for performance
+
+    /**
+     * Initialize AliasAction instance.
+     * @param alias alias name.
+     * @param action instance of concrete Action class.
+     */
+    public AliasAction(
+            @NonNull String alias,
+            @NonNull T action) {
+        if (TextUtils.isEmpty(alias)) {
+            throw new IllegalArgumentException("alias is empty or null");
+        }
+
+        if (action == null) {
+            throw new IllegalArgumentException("action is null");
+        }
         this.alias = alias;
         this.action = action;
     }
 
+    @NonNull
     public String getAlias() {
         return alias;
     }
 
+    @NonNull
     public T getAction() {
         return action;
     }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeString(this.alias);
+        dest.writeSerializable(this.action.getClass());
+        dest.writeString(new Gson().toJson(this.action));
+    }
+
+    public AliasAction(Parcel in) {
+        this.alias = in.readString();
+        Class<T> actionType = (Class<T>)in.readSerializable();
+        String jsonString = in.readString();
+        this.action = new Gson().fromJson(jsonString, actionType);
+    }
+
+    public static final Creator<AliasAction> CREATOR = new Creator<AliasAction>() {
+        @Override
+        public AliasAction createFromParcel(Parcel source) {
+            return new AliasAction(source);
+        }
+
+        @Override
+        public AliasAction[] newArray(int size) {
+            return new AliasAction[size];
+        }
+    };
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null) return false;
+        if (!(o instanceof AliasAction)) return false;
```

Comment:
```java
 if (!(o instanceof AliasAction)) return false;
```

This allows sub class of `AliasAction`. Is it your intention? If No, following code may be better:

```java
if (o.getClass() = AliasAction.class) return false;
```

What do you think?